lib/vuln/php_easter_egg/engine.py

```diff
@@ -0,0 +1,201 @@
+            for extra_requirement in extra_requirements_dict():
+                if extra_requirement in methods_args:
+                    new_extra_requirements[extra_requirement] = methods_args[extra_requirement]
+        extra_requirements = new_extra_requirements
+        if ports is None:
+            ports = extra_requirements["php_easter_eggs_vuln_ports"]
+        if target_type(target) == 'HTTP':
+            target = target_to_host(target)
+        threads = []
+        total_req = len(ports)
+        thread_tmp_filename = '{}/tmp/thread_tmp_'.format(load_file_path()) + ''.join(
+            random.choice(string.ascii_letters + string.digits) for _ in range(20))
+        __log_into_file(thread_tmp_filename, 'w', '1', language)
+        trying = 0
+        keyboard_interrupt_flag = False
+        for port in ports:
+            port = int(port)
+            t = threading.Thread(target=__php_easter_eggs_vuln,
+                                 args=(target, int(port), timeout_sec, log_in_file, language, time_sleep,
+                                       thread_tmp_filename, extra_requirements, socks_proxy, scan_id, scan_cmd))
+            threads.append(t)
+            t.start()
+            trying += 1
+            if verbose_level > 3:
+                info(
+                    messages(language, "trying_message").format(trying, total_req, num, total, target, port, 'php_easter_eggs_vuln_scan'))
+            while 1:
+                try:
+                    if threading.activeCount() >= thread_number:
+                        time.sleep(0.01)
+                    else:
+                        break
+                except KeyboardInterrupt:
+                    keyboard_interrupt_flag = True
+                    break
+            if keyboard_interrupt_flag:
+                break
+        # wait for threads
+        kill_switch = 0
+        kill_time = int(
+            timeout_sec / 0.1) if int(timeout_sec / 0.1) is not 0 else 1
+        while 1:
+            time.sleep(0.1)
+            kill_switch += 1
+            try:
+                if threading.activeCount() is 1:
+                    break
+            except KeyboardInterrupt:
+                break
+        thread_write = int(open(thread_tmp_filename).read().rsplit()[0])
+        if thread_write is 1 and verbose_level is not 0:
+            info(messages(language, "not_found"))
+            data = json.dumps({'HOST': target, 'USERNAME': '', 'PASSWORD': '', 'PORT': '', 'TYPE': 'php_easter_eggs_vuln_scan',
+                               'DESCRIPTION': messages(language, "not_found"), 'TIME': now(),
+                               'CATEGORY': "scan", 'SCAN_ID': scan_id, 'SCAN_CMD': scan_cmd})
+            __log_into_file(log_in_file, 'a', data, language)
+        os.remove(thread_tmp_filename)
+
+    else:
+        print("aman")
+        warn(messages(language, "input_target_error").format(
+            'php_easter_eggs_vuln_scan', target))
```

Comment:
@aman566  should be '_egg' not '_eggs' - please stay consistent and pay attention to details